packages/preview-middleware-client/src/adp/quick-actions/common/op-add-header-field.ts

```diff
@@ -0,0 +1,69 @@
+import OverlayRegistry from 'sap/ui/dt/OverlayRegistry';
+import FlexCommand from 'sap/ui/rta/command/FlexCommand';
+import UI5Element from 'sap/ui/core/Element';
+import ObjectPageLayout from 'sap/uxap/ObjectPageLayout';
+import FlexBox from 'sap/m/FlexBox';
+
+import { SIMPLE_QUICK_ACTION_KIND, SimpleQuickAction } from '@sap-ux-private/control-property-editor-common';
+
+import { DialogNames, handler } from '../../../adp/init-dialogs';
+
+import { getRelevantControlFromActivePage } from '../../../cpe/quick-actions/utils';
+import { QuickActionContext, SimpleQuickActionDefinition } from '../../../cpe/quick-actions/quick-action-definition';
```

Comment:
If quick actions are needed in cpe and adp shouldn't this then be moved to src/utils or somewhere more central at src?

Reply:
The current structure is as follows:
- `flp` code that is required for all previews
- `cpe` RTA plugin for working with CPE
- `adp` extends `cpe` plugin with adaptation specific code

That means `src/utils` probably would need to contain very generic code that would also make sense in scope of `flp`. For things that make sense only in RTA and CPE scope we so far keep in the `cpe` folder.

We probably should move the reuse code a bit higher and make it more clear which part of `cpe` is intended for reuse/extension by `adp`.

Reply:
Thanks for the explanation. This restructuring sounds like a task that does not necessarily need to be done in this PR.